app/src/ui/history/compare.tsx

```diff
@@ -0,0 +1,477 @@
+      nextProps.compareState.formState.kind !==
+      this.props.compareState.formState.kind
+
+    const newFormState = nextProps.compareState.formState
+
+    if (hasFormStateChanged && newFormState.kind === ComparisonView.None) {
+      // the comparison form should be reset to its default state
+      this.setState({ filterText: '', focusedBranch: null })
+      return
+    }
+
+    if (!hasFormStateChanged && newFormState.kind !== ComparisonView.None) {
+      // ensure the filter text is in sync with the comparison branch
+      const branch = newFormState.comparisonBranch
+
+      this.setState({
+        filterText: branch.name,
+        focusedBranch: branch,
+      })
+    }
+
+    if (nextProps.sidebarHasFocusWithin !== this.props.sidebarHasFocusWithin) {
+      if (nextProps.sidebarHasFocusWithin === false) {
+        this.setState({ showBranchList: false })
+      }
+    }
+  }
+
+  public componentWillMount() {
+    this.props.dispatcher.initializeCompare(this.props.repository)
+  }
+
+  public componentWillUnmount() {
+    this.textbox = null
+  }
+
+  public componentDidMount() {
+    if (this.textbox !== null && this.state.showBranchList) {
+      this.textbox.focus()
+    }
+  }
+
+  public render() {
+    const formState = this.props.compareState.formState
+    const placeholderText =
+      formState.kind === ComparisonView.None
+        ? __DARWIN__
+          ? 'Select Branch To Compare...'
+          : 'Select branch to compare...'
+        : undefined
+
+    return (
+      <div id="compare-view">
+        <div className="the-box">
+          <FancyTextBox
+            symbol={OcticonSymbol.gitBranch}
+            type="search"
+            placeholder={placeholderText}
+            onFocus={this.onTextBoxFocused}
+            value={this.state.filterText}
+            onRef={this.onTextBoxRef}
+            onValueChanged={this.onBranchFilterTextChanged}
+            onKeyDown={this.onBranchFilterKeyDown}
+          />
+        </div>
+
+        {this.state.showBranchList
+          ? this.renderFilterList()
+          : this.renderCommits()}
+      </div>
+    )
+  }
+
+  private onBranchesListRef = (branchList: BranchList | null) => {
+    this.branchList = branchList
+  }
+
+  private renderCommits() {
+    const formState = this.props.compareState.formState
+    return (
+      <div className="the-commits">
+        {formState.kind === ComparisonView.None
+          ? this.renderCommitList()
+          : this.renderTabBar(formState)}
+      </div>
+    )
+  }
+
+  private viewHistoryForBranch = () => {
+    this.props.dispatcher.executeCompare(this.props.repository, {
+      kind: CompareActionKind.History,
+    })
+  }
+
+  private renderCommitList() {
+    const compareState = this.props.compareState
+    const selectedCommit = this.state.selectedCommit
+    const commitSHAs = compareState.commitSHAs
+
+    const emptyListMessage =
+      compareState.formState.kind === ComparisonView.None
+        ? 'No history'
```

Comment:
do we want to change the casing based on mac v windows or is that only for buttons and menu items?

Reply:
This is what the current History view shows on macOS when you have no commits:

<img width="331" src="https://user-images.githubusercontent.com/359239/39161345-d7afb17e-4735-11e8-8dfd-ae0ebc63e4c1.png">